Cards/archieve.cpp

```diff
@@ -12954,3 +12980,361 @@ StandardBearer::StandardBearer()
+    };
+}
+
+VandergriftsBlade::VandergriftsBlade()
+{
+    id = "201648";
+    name = "Vandergrift's Blade";
+    text = "Choose One: Destroy a Bronze or Silver Cursed enemy; or Deal 9 damage and, if the unit was destroyed, Banish it.";
+    url = "https://gwent.one/image/card/low/cid/png/" + id + ".png";
+    tags = { Item };
+    isSpecial = true;
+    faction = NothernRealms;
+    rarity = Silver;
+
+    _onPlaySpecial = [=](Field &ally, Field &enemy) {
+        auto *option1 = new VandergriftsBlade::Destroy;
+        copyCardText(this, option1);
+        option1->text = "Destroy a Bronze or Silver Cursed enemy.";
+
+        auto *option2 = new VandergriftsBlade::Damage;
+        copyCardText(this, option2);
+        option2->text = "Deal 9 damage and, if the unit was destroyed, Banish it.";
+
+        _choosen = nullptr;
+        startChoiceToSelectOption(ally, enemy, this, {option1, option2});
+    };
+
+    _onOptionChoosen = [=](Card *target, Field &ally, Field &enemy) {
+        _choosen = target;
+        if (dynamic_cast<VandergriftsBlade::Destroy *>(_choosen))
+            startChoiceToTargetCard(ally, enemy, this, {isBronzeOrSilver, hasTag(Cursed)}, EnemyBoard);
+        if (dynamic_cast<VandergriftsBlade::Damage *>(_choosen))
+            startChoiceToTargetCard(ally, enemy, this, {}, EnemyBoard);
+        assert(false);
+    };
+
+    _onTargetChoosen = [=](Card *target, Field &ally, Field &enemy) {
+        assert(_choosen);
+
+        if (dynamic_cast<VandergriftsBlade::Destroy *>(_choosen)) {
+            putToDiscard(target, ally, enemy, this);
+
+            delete _choosen;
+            _choosen = nullptr;
+            return;
+        }
+
+        if (dynamic_cast<VandergriftsBlade::Damage *>(_choosen)) {
+            if (damage(target, 9, ally, enemy, this))
+                banish(target, ally, enemy, this);
+
+            delete _choosen;
+            _choosen = nullptr;
+            return;
+        }
+
+        assert(false);
+    };
+}
+
+ReinforcedTrebuchet::ReinforcedTrebuchet()
+{
+    id = "122315";
+    name = "Reinforced Trebuchet";
+    text = "Deal 1 damage to a random enemy on turn end.";
+    url = "https://gwent.one/image/card/low/cid/png/" + id + ".png";
+    tags = { Machine };
+    power = powerBase = 8;
+    faction = NothernRealms;
+    rarity = Bronze;
+
+    _onTurnEnd = [=](Field &ally, Field &enemy) {
+        damage(random(cardsFiltered(ally, enemy, {}, EnemyBoard), ally.rng), 1, ally, enemy, this);
+    };
+}
+
+Ballista::Ballista()
+{
+    id = "122301";
+    name = "Ballista";
+    text = "Deal 1 damage to an enemy and 4 other random enemies with the same power. Crewed: Repeat its ability.";
+    url = "https://gwent.one/image/card/low/cid/png/" + id + ".png";
+    tags = { Machine };
+    power = powerBase = 6;
+    faction = NothernRealms;
+    rarity = Bronze;
+
+    _onDeploy = [=](Field &ally, Field &enemy) {
+        int n = nCrewed(this, ally);
```

Comment:
```suggestion
        int n = 1 + nCrewed(this, ally);
```
наверное на 1 больше, тк число nCrewed может быть 0

Reply:
Переделала метод nCrewed, теперь он действительно может быть 0